package.json

```diff
@@ -18,13 +18,15 @@
     "test:codecov-upload": "nyc report --reporter=json && codecov -f coverage-report/*.json",
     "sonar-scanner": "sonar-scanner -Dsonar.projectName='CMC :: Citizen Frontend' -Dsonar.sources=src/main -Dsonar.tests=src/test -Dsonar.exclusions=src/main/public/** -Dsonar.typescript.lcov.reportPaths=coverage-report/lcov.info",
     "precommit": "yarn lint",
-    "prepush": "yarn test && yarn test:routes"
+    "prepush": "yarn test && yarn test:routes",
+    "snyk-protect": "snyk protect --org=hmcts --debug",
+    "prepare": "yarn snyk-protect"
```

Comment:
How much time `snyk-protect` takes? Will that affect build time?

Reply:
It's around 20 seconds @geek-soft. Perhaps we could skip it for testing but it's mandatory for RPM builds -- otherwise the service shipped in RPM will not be patched for security vulnerabilities.

Reply:
the snyk thing doesn't patch it though, it just adds a reminder to check again later at a certain date.

Reply:
Having this as pre-install script I think will execute scan at least twice (at host build time + docker image build time). I even think it might get executed 4 times since we first install all dependencies (including dev ones) and then we only install production dependencies.

If that is the case it would mean 4x20 which is a lot. Not doubt snyk should get executed in the pipeline, just one as pre installation step.

Reply:
Hmm, it looks like `prepare` script is not executed on `yarn install --production` or `npm install  --production` so we are down to 2 snyk scans.

Reply:
Nope, we are doing it 3 times. One as first build step, second one within docker and third one for RPM.

Reply:
@geek-soft the hook can be removed and we can call it explicitly before packaging the RPM.

> the snyk thing doesn't patch it though

@timja-kainos https://snyk.io/docs/using-snyk/#protect